radicle/src/storage/git/transport/local/url.rs

```diff
@@ -0,0 +1,106 @@
+//! Git local transport URLs.
+use std::str::FromStr;
+
+use thiserror::Error;
+
+use crate::{
+    crypto,
+    identity::{Id, IdError},
+};
+
+/// Repository namespace.
+type Namespace = crypto::PublicKey;
+
+#[derive(Debug, Error)]
+pub enum UrlError {
+    /// Invalid format.
+    #[error("invalid url format: expected `rad://<repo>[/<namespace>]`")]
+    InvalidFormat,
+    /// Unsupported URL scheme.
+    #[error("unsupported scheme: expected `rad://`")]
+    UnsupportedScheme,
+    /// Invalid repository identifier.
+    #[error("repo: {0}")]
+    InvalidRepository(#[source] IdError),
+    /// Invalid namespace.
+    #[error("namespace: {0}")]
```

Comment:
Adding either ' or " around the values would make it really clear if the invalid namespace or repo is empty or has empty spaces.  What do you think?